File: day-of-the-programmer/DayOfTheProgrammer.java
import java.io.*;

class Result {

    /*
     * Complete the 'dayOfProgrammer' function below.
     *
     * The function is expected to return a STRING.
     * The function accepts INTEGER year as parameter.
     */

    public static String dayOfProgrammer(int year) {
        // Write your code here
        if (year == 1918)
            return "26.09.1918";

        int day = 13;
        // gregorian calendar
        if (year > 1918 && (year % 400 == 0 || (year % 4 == 0 && year % 100 != 0)))
            day = 12;

        // julian calendar
        else if (year <= 1917 && year % 4 == 0)
            day = 12;

        return day + ".09." + year;
    }

}

public class DayOfTheProgrammer {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        int year = Integer.parseInt(bufferedReader.readLine().trim());

        String result = Result.dayOfProgrammer(year);

        bufferedWriter.write(result);
        bufferedWriter.newLine();

        bufferedReader.close();
        bufferedWriter.close();
    }
}

File: day-of-the-programmer/day-of-the-programmer.js
"use strict";

const fs = require("fs");

process.stdin.resume();
process.stdin.setEncoding("utf-8");

let inputString = "";
let currentLine = 0;

process.stdin.on("data", function (inputStdin) {
  inputString += inputStdin;
});

process.stdin.on("end", function () {
  inputString = inputString.split("\n");

  main();
});

function readLine() {
  return inputString[currentLine++];
}

/*
 * Complete the 'dayOfProgrammer' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts INTEGER year as parameter.
 */

function dayOfProgrammer(year) {
  // Write your code here
  if (year == 1918) return "26.09.1918";

  let day = 13;
  // gregorian calendar
  if (year > 1918 && (year % 400 == 0 || (year % 4 == 0 && year % 100 != 0)))
    day = 12;
  // julian calendar
  else if (year <= 1917 && year % 4 == 0) day = 12;

  return day + ".09." + year;
}

function main() {
  const ws = fs.createWriteStream(process.env.OUTPUT_PATH);

  const year = parseInt(readLine().trim(), 10);

  const result = dayOfProgrammer(year);

  ws.write(result + "\n");

  ws.end();
}
